private/buf/bufsync/syncer.go

```diff
@@ -629,6 +653,142 @@ func (s *syncer) readModuleAt(
 	return builtModule, nil
 }
 
+// attachOlderTags takes syncable commits for a branch already calculated, and looks back for each
+// module a given amount of commits or timestamps, syncing tags in case they were created or moved
+// after those commits were synced.
+func (s *syncer) attachOlderTags(ctx context.Context, branch string, syncableCommits []*syncableCommit) error {
+	branchModulesForSync, ok := s.branchesToModulesForSync[branch]
+	if !ok || len(branchModulesForSync) == 0 {
+		// branch should not be synced, or no modules to sync in that branch
+		return nil
+	}
```

Comment:
Should this check be done one layer up in `syncBranch`?

Reply:
We can add it there, yes, although I think it's no harm to add it as well in the inner functions, in case those are reused sometime later by another func.

Reply:
Ah yeah, I see -- and checking the map is pretty cheap. I think this is reasonable :)